sys/ztimer/auto_init.c

```diff
@@ -52,78 +53,206 @@
 #define FREQ_1MHZ       1000000LU
 #define FREQ_250KHZ     250000LU
 #define FREQ_1KHZ       1000LU
+#define FREQ_1HZ        1LU
+
+
+/* Step 0: define available ztimer-periphery by activated modules */
+
+/* #if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+ * replaces #if MODULE_ZTIMER_PERIPH_TIMER
+ * the ztimer_periph_timer is always available
+ * having an (1) config defined in ztimer/config.h
+ */
+
+#if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+#  define ZTIMER_TIMER      _ztimer_periph_timer
+#  define ZTIMER_TIMER_CLK  _ztimer_periph_timer.super
+#endif
+
+#if MODULE_ZTIMER_PERIPH_RTT
+#  define ZTIMER_RTT      _ztimer_periph_timer_rtt
+#  define ZTIMER_RTT_CLK  _ztimer_periph_timer_rtt
+#endif
+
+#if MODULE_ZTIMER_PERIPH_RTC
+#  define ZTIMER_RTC      _ztimer_periph_timer_rtc
+#  define ZTIMER_RTC_CLK  _ztimer_periph_timer_rtc
+#endif
+
+/* Step 1: prepare defines for ztimer initialization and
+ *         check which ztimer-periphery is in use
+ */
 
 #if MODULE_ZTIMER_USEC
-#  if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
-static ztimer_periph_timer_t _ztimer_periph_timer_usec = {
+#  ifndef INIT_ZTIMER_TIMER
+#    define INIT_ZTIMER_TIMER 1
+#  endif
+#endif
+
+#if MODULE_ZTIMER_MSEC
+#  if defined(ZTIMER_RTT) && RTT_FREQUENCY >= FREQ_1KHZ
```

Comment:
Isn't there frequency shifting as well?

Reply:
The rtt-frequency is set by the rtt initialization which is mcu dependent, if it can be set, a number of boards default to 1Hz (see # #16116 i think this should be changend but this is not part of this PR) therefore the Test for the rtt-frequency. @maribu recently made the sugestion in #16116 that periph timer might be useable for rtt-/lp-timers which would make the configuration availible to ztimer.

Reply:
Now that I think of it, shifting only works for power of two frequencies., so this is fine.